ElvargServer/src/main/java/com/elvarg/game/entity/impl/player/Player.java

```diff
@@ -503,6 +506,19 @@ public void process() {
 				}
 			}
 		}
+
+		if (isAfk()) {
+			PLAYER_PERSISTENCE.save(this);
+			requestLogout();
+		}
+	}
+
+	public void setIdleTime() {
+		idleTime = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(5);
+	}
+
+	public boolean isAfk() {
+		return idleTime < System.currentTimeMillis() && isOnline && this.getRights().ordinal() < 2;

```

Comment:
I'm a bit iffy about `isOnline`. why not like this?
```suggestion
		return idleTime != 0 && idleTime < System.currentTimeMillis() && this.getRights().ordinal() < 2;

```

Reply:
I tend to use isOnline for future use It will come in handy for countless actions.

Reply:
Can you list them?

I rather would have `isOnline` as a function that checks if we are or not based on if socket session is still active.